R/helper.R

```diff
@@ -0,0 +1,43 @@
+#' Retrieve SingleR reference datasets from Github
+#'
+#' \code{getSinglerRef} retrieves \code{SingleR} reference datasets from Github. 
+#'
+#' @param dataset String referring to SingleR reference dataset to retrieve. 
+#'     Options include:
+#'     \itemize{
+#'       \item "hpca"
+#'       \item "blueprint_encode"
+#'       \item "immgen"
+#'       \item "mouse.rnaseq"
+#'     }
+#' @return SingleR reference dataset.
+#'
+#' @author Jared Andrews
+#'
+#' @examples
+#'
+#' immgen <- getSinglerRef(dataset = "immgen")
+#' mouse.rnaseq <- getSinglerRef(dataset = "mouse.rnaseq")
+#'
+#' @export
+#' @importFrom curl curl_download
+#' 
+getSinglerRef <- function(dataset = "hpca") {
```

Comment:
Move `opts` as the default `dataset=` argument and use `dataset <- match.arg(dataset)` to do the check.